src/span.js

```diff
@@ -168,6 +179,19 @@ export default class Span {
     }
 
     /**
+     * Checks whether a span's sampling decision is finalized by calling setOperationName 
+     * This is used to hold onto tags, and logs, so that we can re-sample the span after
+     * an operation name is set on the span.  This is necessary because setting a sampling
+     * rate on the span before we know the operation name means adaptive-sampling
+     * does not know how to correlate the operation with sampling decisions.
+     *
+     * @return {boolean} - The decision about whether this span is done being sampled.
+     **/
+    isSampledAndFinished(): boolean {
+        return this._spanContext.isSampled() || !this._spanContext.samplingFinished;
```

Comment:
this logic is more like: is SampledOrFinished. even then !this._spanContext.samplingFinished is true when samplingFinished is false no?